igor-web/src/main/groovy/com/netflix/spinnaker/igor/build/BuildController.groovy

```diff
@@ -171,6 +172,23 @@ class BuildController {
     "true"
   }
 
+  @RequestMapping(value = "/masters/{name}/jobs/**/update/{buildNumber}", method = RequestMethod.PATCH)
+  @PreAuthorize("hasPermission(#master, 'BUILD_SERVICE', 'WRITE')")
+  void update(
+    @PathVariable("name") String master,
+    @PathVariable("buildNumber") Integer buildNumber,
+    @RequestBody UpdatedBuild updatedBuild,
+    HttpServletRequest request
+  ) {
+    def jobName = ((String) request.getAttribute(HandlerMapping.PATH_WITHIN_HANDLER_MAPPING_ATTRIBUTE))
```

Comment:
this is wild and crazy

I get that it is the pattern everywhere else in here though..

Reply:
Totally agree on the wild and craziness.